### ISSUE TITLE
Incorrect example in `configure_optimizers`

### ISSUE THREAD

**Issue by lucadiliello** [PyTorchLightning/pytorch-lightning#5066]
## 🐛 Bug

Example shows to use `scheduler` key to pass the scheduler in the dict. Correct key is `lr_scheduler` instead.

The example is:
```python
{
    'scheduler': lr_scheduler, # The LR schduler
    'interval': 'epoch', # The unit of the scheduler's step size
    'frequency': 1, # The frequency of the scheduler
    'reduce_on_plateau': False, # For ReduceLROnPlateau scheduler
    'monitor': 'val_loss', # Metric for ReduceLROnPlateau to monitor
    'strict': True # Whether to crash the training if `monitor` is not found
}
```

but should be 

```python
{
    'lr_scheduler': lr_scheduler, # The LR schduler
    'interval': 'epoch', # The unit of the scheduler's step size
    'frequency': 1, # The frequency of the scheduler
    'reduce_on_plateau': False, # For ReduceLROnPlateau scheduler
    'monitor': 'val_loss', # Metric for ReduceLROnPlateau to monitor
    'strict': True # Whether to crash the training if `monitor` is not found
}
```

### To Reproduce
Just use the wrong `scheduler` key. No exception will be raised but no scheduler will be used! I think many people, like me, do not know that their scheduler is not running. I would add at least some warning.

### Environment

 - PyTorch Version (e.g., 1.7.1):
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.8.5
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by rohitgr7** [comment]
no it's correct. If you'll use `'lr_scheduler'` key, it will raise an exception.

**Comment by lucadiliello** [comment]
Double checked on master:
1) See `pytorch_lightning.trainer.optimizers` -> `TrainerOptimizersMixin.init_optimizers` line 55
2) Using `lr_scheduler` does log LRs, whitout nothing is logged and it get the warning: `RuntimeWarning: You are using LearningRateMonitor callback with models that have no learning rate schedulers. Please see documentation for configure_optimizers method.`

**Comment by rohitgr7** [comment]
mind reproduce it with https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing?

**Comment by lucadiliello** [comment]
it is difficult to reproduce it in Colab since I do not know where the value of learning rates is stored and I cannot check whether it is changing or not. I noticed it from the `tensorboard` plots.

**Comment by rohitgr7** [comment]
can you put your `configure_optimizer` code here? I'll try with master/

**Comment by lucadiliello** [comment]
Closing since the issue was mine misinterpretation of the doc. I was convinced that the given example was about the return of the `configure_optimizers` method but was wrt the `lr_dict` instead. Sorry for bothering you.
